features/fixtures/mazerunner/cxx-scenarios/src/main/java/com/bugsnag/android/mazerunner/scenarios/CXXSessionInfoCrashScenario.kt

```diff
@@ -0,0 +1,29 @@
+package com.bugsnag.android.mazerunner.scenarios
+
+import android.content.Context
+import com.bugsnag.android.Bugsnag
+import com.bugsnag.android.Configuration
+
+class CXXSessionInfoCrashScenario(
+    config: Configuration,
+    context: Context,
+    eventMetadata: String?
+) : Scenario(config, context, eventMetadata) {
+
+    init {
+        System.loadLibrary("cxx-scenarios")
+        config.autoTrackSessions = false
+    }
+
+    external fun crash(value: Int): Int
+
+    override fun startScenario() {
+        super.startScenario()
+        if (eventMetadata!! != "non-crashy") {
+            Bugsnag.startSession()
+            Bugsnag.notify(Exception("For the first"))
+            Bugsnag.notify(Exception("For the second"))
```

Comment:
Are calls to `notify` synchronous?  I.e. crashing immediately afterwards will not cause flakes?

Reply:
The error capture is synchronous but the delivery is asynchronous. In practice I don't think it will be an issue as the JNI call should take some time to execute, by which time the delivery should be done.

If it is an issue on the full CI run, or if it crops up as a failure, then it should be a easy enough fix of adding an `InterceptingDelivery` implementation that only calls `crash()` once the two error reports are sent.

Reply:
Updated to use `InterceptingDelivery` as this was indeed causing flakiness.